lib/libzfs/libzfs_dataset.c

```diff
@@ -4107,6 +4107,13 @@ zfs_promote(zfs_handle_t *zhp)
 
 	if (ret != 0) {
 		switch (ret) {
+		case EACCES:
+			/* There is a conflicting snapshot name. */
```

Comment:
Looks like you accidentally brought this comment along from below and it should be removed.